core/commands/storage/path.go

```diff
@@ -0,0 +1,163 @@
+package storage
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/TRON-US/go-btfs-cmds"
+	"github.com/mitchellh/go-homedir"
+	"github.com/prometheus/common/log"
+)
+
+const (
+	storeDir = ".btfs"
+	fileName = "path.properties"
+)
+
+var btfsPath string
+var filePath string
+var StorePath string
+var OriginPath string
+
+var PathCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Modify the Host storage folder path for BTFS client.",
+		ShortDescription: `
+The default local repository path is located at ~/.btfs folder, in order to
+improve the hard disk space usage, provide the function to change the original 
+storage location, a specified path as a parameter need to be passed.
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("path-name", true, true, "New BTFS Path.").EnableStdin(),
+	},
+
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		StorePath = strings.Trim(req.Arguments[0], " ")
+
+		if StorePath == "" {
+			return fmt.Errorf("path is not defined")
+		}
+
+		defaultPath := "~/.btfs"
+		if btfsPath != "" {
+			if btfsPath != StorePath {
+				OriginPath = filepath.Join(btfsPath, storeDir)
+			} else {
+				return fmt.Errorf("specifed path is same with current path")
+			}
+		} else if home, err := homedir.Expand(defaultPath); err == nil && home != "" {
+			OriginPath = home
+		} else {
+			return fmt.Errorf("can not find the original stored path")
+		}
+
+		if CheckExist(StorePath) == false {
+			err := os.MkdirAll(StorePath, os.ModePerm)
+			if err != nil {
+				return fmt.Errorf("mkdir: %s", err)
+			}
+		} else if CheckDirEmpty(filepath.Join(StorePath, storeDir)) == false {
+			return fmt.Errorf("path is occupied")
+		}
+
+		restartCmd := exec.Command("btfs", "restart")
+		if err := restartCmd.Run(); err != nil {
+			return fmt.Errorf("restart command: %s", err)
+		}
+		return nil
+	},
+}
+
+func init() {
+	GetPropertiesPath()
+	SetEnvVariables()
+}
+
+func WriteProperties() error {
+	if CheckExist(filePath) == false {
+		newFile, err := os.Create(filePath)
+		defer newFile.Close()
+		if err != nil {
+			return err
+		}
+	}
+	data := []byte(StorePath)
+	err := ioutil.WriteFile(filePath, data, 0666)
+	if err == nil {
+		fmt.Printf("Storage location was reset in %v\n", StorePath)
+	}
+	return err
+}
+
+func MoveFolder() error {
+	// make dir does not contain .btfs, but move need to specify .btfs
+	err := os.Rename(filepath.Join(OriginPath), filepath.Join(StorePath, storeDir))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func ReadProperties(filePath string) string {
+	f, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		log.Errorf("Read properties fail: [%v]\n", err)
+	}
+	return string(f)
+}
+
+func CheckDirEmpty(dirname string) bool {
+	dir, _ := ioutil.ReadDir(dirname)
```

Comment:
handle err here

Reply:
ioutil.ReadDir(dirname) will produce an error if the .btfs folder is not existed, if we handle the error here, we can not detect .btfs folder is not in the target folder, only added log here for the err.